packages/adblocker/src/filters/network.ts

```diff
@@ -400,18 +397,171 @@ function compileRegex(
+
+    if (code === 61 /* '=' */ || code === 44 /* ',' */) {
+      end = pos;
+
+      break;
+    }
+  }
+
+  return [pos, line.slice(start, end)] as const;
+}
+
+/**
+ * Collects a filter option value until the function sees the special character.
+ * This function respects the escaping characters, so we can safely collect the full value
+ * including the special characters which are not allowed normally.
+ * This function will stop if it sees a comma sign.
+ */
+function getFilterOptionValue(line: string, pos: number, end: number): [number, string] {
+  let start = pos;
+  let value = '';
+
+  for (let code: number; pos < end; pos++) {
+    code = line.charCodeAt(pos);
+
+    if (code === 92 /* '\\' */) {
+      value += line.slice(start, pos);
+      start = ++pos;
+    } else if (code === 44 /* ',' */) {
+      break;
+    }
+  }
+
+  if (start - pos !== 0) {
+    value += line.slice(start, pos);
+  }
+
+  return [pos, value];
+}
+
+/**
+ * Collects a filter option value of the replace modifier.
+ * This function respects the escaping character with the allowed characters of the replace modifier.
+ * In the replace modifier, it can include the any sign allowed in the regular expression.
+ * Therefore, a comma sign can interfere the `getFilterOptionValue` function.
+ * This function will not stop unless it collects the all of parts of the replace modifier option value.
+ */
+function getFilterReplaceOptionValue(line: string, pos: number, end: number): [number, string[]] {
+  // Try to fast exit if the first character is an unexpected character.
+  if (line.charCodeAt(pos++) !== 47 /* '/' */) {
+    return [end, []];
+  }
+
+  const parts = ['', '', ''];
+
+  let start = pos;
+  let slashes = 0;
+
+  for (let code: number; pos < end; pos++) {
+    code = line.charCodeAt(pos);
+
+    if (code === 92 /* '\\' */) {
+      parts[slashes] += line.slice(start, pos);
+      start = pos;
+    } else if (code === 47 /* '/' */) {
+      if (pos - start !== 0) {
+        parts[slashes] += line.slice(start, pos);
+      }
+
+      start = pos + 1;
+
+      if (++slashes === 2) {
+        // Since we saw 3 slashes in total, it means that the option value should be closed here.
+        // Note that we already saw the first slash before the loop.
+        break;
+      }
+    }
+  }
+
+  const valueEnd = line.indexOf(',', pos);
+
+  if (valueEnd !== -1) {
+    end = valueEnd;
+  }
+
+  parts[2] = line.slice(start, end);
+
+  pos = end;
+
+  return [pos, parts];
+}
+
+/**
+ * Collects an array of filter options from the given index.
+ * This function leverages `getFilterOptionKey`, `getFilterOptionValue`, and every extension functions.
+ * Depending on the filter option key, the function to collect filter option value can vary.
+ * For the generic filter option value, it'll use `getFilterOptionValue` function to get the value.
+ */
+function getFilterOptions(line: string, pos: number, end: number) {
+  const options: Array<[string, string]> = [];
```

Comment:
What if there is no value? Maybe the type could be like this instead?
```suggestion
  const options: Array<[string, string | undefined]> = [];
```

Reply:
Since we're slicing all strings, there'll be empty string instead of undefined.